src/main/java/software/amazon/awssdk/crt/s3/S3MetaRequest.java

```diff
@@ -38,6 +38,12 @@ protected boolean canReleaseReferencesImmediately() {
     @Override
     protected void releaseNativeHandle() {
         if (!isNull()) {
+            /**
+             * Cancel the meta request before drop the refcount.
+             * The meta request is not referenced by Java any longer, everything from native to Java will be ignored.
+             * Cancelling the meta request instead of letting it keep flowing.
+             **/
+            s3MetaRequestCancel(getNativeHandle());
```

Comment:
What happens if the request is already cancelled or paused?

Reply:
it won't do anything. The [cancel](https://github.com/awslabs/aws-c-s3/blob/main/source/s3_meta_request.c#L351-L358) is idempotent, and the [pause](https://github.com/awslabs/aws-c-s3/blob/main/source/s3_auto_ranged_put.c#L1668-L1677) also fall into the same path with cancel.